web/src/buddy_list.js

```diff
@@ -97,45 +408,161 @@ export class BuddyList extends BuddyListConf {
         this.update_padding();
     }
 
+    should_hide_headers(current_sub, pm_ids_set) {
+        // If we have only "other users" and aren't in a stream/DM view
+        // then we don't show section headers and only show one untitled
+        // section.
+        return this.users_matching_view_ids.length === 0 && !current_sub && !pm_ids_set.size;
+    }
```

Comment:
Hmm, why is this the right check? I guess it would seem to me like the check should be just `!current_sub && !pm_ids_set.size`, without checking `this.users_matching_view_ids.length === 0`, which is by definition going to be true if the other things are true?

Reply:
Ah true! I think this happened because I started with just `this.users_matching_view_ids.length === 0` and adjusted it later. Thanks for noticing that. I'll edit the `Calculate render data once per view.` commit to calculate `should_hide_headers` that way.